bin/oconf

```diff
@@ -1,80 +1,52 @@
-        type: 'string'
-	})
-    .option('option-as-json', {
-        alias: 'j',
-        description: 'The extracted value will be extracted as JSON.',
-        type: 'boolean'
-    })
-    .option('public', {
-        alias: 'p',
-        description: 'Only values in #public objects will be extracted.',
-        type: 'boolean'
-    })
-    .option('json', { type: 'boolean' })
-    .option('allow-missing-option', {
-        alias: 'm',
-        description: 'Do not fail when no such value is present.',
-        type: 'boolean'
-    })
-    .option('allowmissing', { type: 'boolean' });
-
-var argv = argp.argv;
+const oconf = require('../lib/index');
+const { Command } = require('commander');
+const program = new Command();
+
+program
+    .name('oconf')
+    .usage('[--lint|--extract-option <option> [--allow-missing-option][--option-as-json]] <file>')
+    .option('-h, --help', 'This help text.')
+    .option('-i, --ignore <path>', 'Absolute path or glob pattern for ignoring included files.')
+    .option('-l, --lint', 'Lint the input file and report any errors.')
+    .option('-r, --relaxed', 'Use relaxed JSON format.')
+    .option('-p, --public', 'Only values in #public objects will be extracted.')
+    .option('-o, --extract-option <option>', 'Extract the value of an option.')
+    .option('-j, --option-as-json', 'The extracted value will be extracted as JSON.')
+    .option('-m, --allow-missing-option', 'Do not fail when no such value is present.')
+    .option('--json', 'JSON output.')
+    .option('--allowmissing', 'Allow missing options.')
+    .parse();
+
+const argv = program.opts();
+argv._ = program.args;
 
 // This is needed as optimist will not force exit when --help is
 // passed on it's own, and we want it to.
 if (argv.help || argv._.length < 1) {
-    argp.showHelp();
+    program.help();
     process.exit(0);
 }
 
 //////////////////////////////////////////////////////////////////
 // This is here for backwards compatibility. Used to be an okay
 // flag. We can add a deprecation warning later.
 if (argv['json']) {
-    argv['option-as-json'] = true;
+    argv['optionAsJson'] = true;
 }
 if (argv['allowmissing']) {
-    argv['allow-missing-option'] = true;
+    argv['allowMissingOption'] = true;
 }
 //////////////////////////////////////////////////////////////////
 
-if (!argv['extract-option'] && (argv['option-as-json'] || argv['allow-missing-option'])) {
-    if (argv['option-as-json']) {
+if (!argv['extractOption'] && (argv['optionAsJson'] || argv['allowMissingOption'])) {
+    if (argv['optionAsJson']) {
         console.error('The flag --option-as-json does not make sense without the --extract-option flag.');
-    } else if (argv['allow-missing-option']) {
+    } else if (argv['allowMissingOption']) {
         console.error('The flag --allow-missing-option does not make sense without the --extract-option flag.');
     }
 
-    console.error('');
-    argp.showHelp();
+    program.help();
```

Comment:
We can update it to `program.outputHelp()` so that it doesn't exit with success. If there was a related test which might have been updated, then that can also be fixed back.